test/Libraries/Microsoft.Extensions.AI.Tests/Image/OpenTelemetryImageGeneratorTests.cs

```diff
@@ -0,0 +1,163 @@
+            OriginalImages = [new UriContent("http://example/input.png", "image/png")],
+        };
+
+        ImageGenerationOptions options = new()
+        {
+            Count = 2,
+            ImageSize = new(1024, 768),
+            MediaType = "image/jpeg",
+            ModelId = "mycoolimagemodel",
+            AdditionalProperties = new()
+            {
+                ["service_tier"] = "value1",
+                ["SomethingElse"] = "value2",
+            },
+        };
+
+        await g.GenerateAsync(request, options);
+
+        var activity = Assert.Single(activities);
+
+        Assert.NotNull(activity.Id);
+        Assert.NotEmpty(activity.Id);
+
+        Assert.Equal("localhost", activity.GetTagItem("server.address"));
+        Assert.Equal(12345, (int)activity.GetTagItem("server.port")!);
+
+        Assert.Equal("generate_content mycoolimagemodel", activity.DisplayName);
+        Assert.Equal("testservice", activity.GetTagItem("gen_ai.provider.name"));
+
+        Assert.Equal("mycoolimagemodel", activity.GetTagItem("gen_ai.request.model"));
+        Assert.Equal(2, activity.GetTagItem("gen_ai.request.choice.count"));
+        Assert.Equal(1024, activity.GetTagItem("gen_ai.request.image.width"));
+        Assert.Equal(768, activity.GetTagItem("gen_ai.request.image.height"));
+        Assert.Equal(enableSensitiveData ? "value1" : null, activity.GetTagItem("service_tier"));
+        Assert.Equal(enableSensitiveData ? "value2" : null, activity.GetTagItem("SomethingElse"));
+
+        Assert.Equal(10, activity.GetTagItem("gen_ai.usage.input_tokens"));
+        Assert.Equal(20, activity.GetTagItem("gen_ai.usage.output_tokens"));
+
+        Assert.True(activity.Duration.TotalMilliseconds > 0);
+
+        var tags = activity.Tags.ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
+        if (enableSensitiveData)
+        {
+            Assert.Equal(ReplaceWhitespace("""
+                [
+                  {
+                    "role": "user",
+                    "parts": [
+                      {
+                        "type": "text",
+                        "content": "This is the input prompt."
+                      },
+                      {
+                        "type": "image",
+                        "content": "http://example/input.png"
+                      }
+                    ]
+                  }
+                ]
+                """), ReplaceWhitespace(tags["gen_ai.input.messages"]));
+
+            Assert.Equal(ReplaceWhitespace("""
+                [
+                  {
+                    "role": "assistant",
+                    "parts": [
+                      {
+                        "type": "image",
+                        "content": "http://example/output.png"
+                      },
+                      {
+                        "type": "image",
+                        "content": "data:image/png;base64,AQIDBA=="
+                      }
```

Comment:
Is the image part from a standard, is it under discussion? Could you point me towards where it is being discussed? I don't see it in the current schema: https://opentelemetry.io/docs/specs/semconv/gen-ai/gen-ai-output-messages.json

Aspire GenAI viewer could be improved to display the image. Either have a src with the URL, or use the base64 data.

Reply:
> Is the image part from a standard

Nope. It has "generate_content" and "image" defined in https://opentelemetry.io/docs/specs/semconv/gen-ai/gen-ai-spans/, and it has the "GenericPart" which this is using and which is documented as "Represents an arbitrary message part with any type and properties.\nThis allows for extensibility with custom message part types.". Basically I'm just trying to piggyback on what's already there for chat and embeddings.

Reply:
Ok.

At a minimum I'll improve the GenAI viewer so it displays the attributes for generic parts. I don't think it displays anything at the moment.

Maybe then add some special case code in case the part has `type="image"`. If a standard springs up in the future then we can both move towards it. Could you create an issue on the OTEL semantic conventions about image parts if it does already exist? It would be nice if it was standardized.